libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/api/MapboxRouteLineApi.kt

```diff
@@ -1445,14 +1302,23 @@ class MapboxRouteLineApi(
             }
         }
 
-        // If the displayRestrictedRoadSections is true then produce a gradient that is transparent
-        // except for the restricted sections. If false produce a gradient for the restricted
-        // line layer that is completely transparent.
+        // If the displayRestrictedRoadSections is true AND the route has restricted sections
+        // then produce a gradient that is transparent except for the restricted sections.
+        // If false produce a gradient for the restricted line layer that is completely transparent.
         val primaryRouteRestrictedSectionsExpressionDef = jobControl.scope.async {
             partitionedRoutes.first.firstOrNull()?.route?.run {
                 if (routeLineOptions.displayRestrictedRoadSections) {
-                    MapboxRouteLineUtils.getRestrictedLineExpressionProducer(
-                        this.directionsRoute,
+                    val extractedRouteData =
+                        extractRouteRestrictionData(this).also {
+                            jobControl.scope.launch(Dispatchers.Main) {
+                                routeHasRestrictions = it.asSequence().filter {
```

Comment:
I was on the fence about removing it.  I wanted to make sure there was no cost to determining the result since this is read as part of the vanishing route line calculation.  I removed it.